File: src/BiddingService/Entities/Bid.cs
using BiddingService.Enums;

using MongoDB.Entities;

namespace BiddingService.Entities;

public class Bid : Entity
{
  public string    AuctionId { get; set; }

  public string    Bidder    { get; set; }

  public DateTime  BidTime   { get; set; } = DateTime.UtcNow;

  public int       Amount    { get; set; }

  public BidStatus BidStatus { get; set; }
}

File: frontend/client/src/types/Bid.ts
export enum bidStatus {
    ACCEPTED = 'Accepted',
    ACCEPTED_BELOW_RESERVE = 'AcceptedBelowReserve',
    TOO_LOW = 'TooLow'
}

export type Bid = {
    id: string
    auctionId: string
    bidder: string
    bidTime: string
    amount: number
    bidStatus: bidStatus
}